### PR TITLE
fix(rds): use MySQL 8.4 auth options

### LINKED ISSUE #2768: RDS: MySQL 8.4 engine container started with --default-authentication-plugin, removed in 8.4 — engine exits(1) while instance reports "available"
Floci version: 1.7.0

CreateDBInstance with engine=mysql, engine_version=8.4 starts a mysql:8.4 container with CMD ["--default-authentication-plugin=mysql_native_password"]. That option was deprecated in MySQL 8.0 and removed in 8.4, so the engine aborts during init:

[ERROR] [MY-000067] unknown variable 'default-authentication-plugin=mysql_native_password'.
[ERROR] [MY-010119] Aborting

The container exits(1), yet DescribeDBInstances reports the instance as available and the advertised endpoint accepts TCP connections and then drops them ("Lost connection to MySQL server") — very confusing to debug, since everything looks healthy from the API.

Repro: create any MySQL 8.4 DB instance (e.g. Terraform aws_db_instance), then docker ps -a | grep floci-rds → Exited (1).

Expected: don't pass the flag for engine ≥ 8.4 (or at all), and reflect a dead engine container in the instance status instead of available.

### PR DESCRIPTION
**Pull request by aniketshukla1**
## Summary

Fix MySQL 8.4 RDS containers failing at startup because the removed default-authentication-plugin option was still passed to the server.

- detect numeric MySQL image tags, including registry ports, patch tags, and digest suffixes
- enable mysql_native_password and select it through authentication-policy on MySQL 8.4
- preserve the existing startup option for MySQL versions through 8.3
- let MySQL 9+ and unversioned images use the server-advertised authentication plugin
- validate both mysql_native_password and caching_sha2_password master scrambles in the RDS proxy
- add regression coverage with a fixed caching-SHA2 protocol vector

Closes #2768

## Type of change

- [x] Bug fix (fix:)
- [ ] New feature (feat:)
- [ ] Breaking change (feat!: or fix!:)
- [ ] Docs / chore

## AWS Compatibility

MySQL 8.4 removed default-authentication-plugin and requires mysql_native_password to be enabled separately. Explicit 8.4 images use those supported options. MySQL 9 removed native-password support entirely, so 9+ and indeterminate image tags use their advertised default and the proxy validates caching-SHA2 authentication directly.

The focused RDS container, service, and MySQL protocol suites pass locally: 289 tests, 0 failures.

## Checklist

- [ ] ./mvnw test passes locally
- [ ] New or updated integration test added
- [x] Commit messages follow Conventional Commits

Focused verification: ./mvnw test -Dtest=RdsContainerManagerTest,RdsServiceTest,MySqlProtocolHandlerTest

### CODE REVIEW

File: src/main/java/io/github/hectorvent/floci/services/rds/container/RdsContainerManager.java [hunk 1]
```diff
@@ -667,12 +667,52 @@ private List<String> buildEnvVars(DatabaseEngine engine, String masterUsername,
...
+        while (minorEnd < tag.length() && Character.isDigit(tag.charAt(minorEnd))) {
+            minorEnd++;
+        }
+        if (minorEnd == dot + 1) {
+            return false;
+        }
+        try {
+            int major = Integer.parseInt(tag.substring(0, dot));
+            int minor = Integer.parseInt(tag.substring(dot + 1, minorEnd));
+            return major == 8 && minor >= 4;
```
[the remark this answers is not in the document]
Comment (aniketshukla1) [reply src/main/java/io/github/hectorvent/floci/services/rds/container/RdsContainerManager.java]: Fully addressed in ee52d985. MySQL 9 and unversioned references now receive no version-specific startup flag and use the server-advertised authentication plugin. The proxy parses that plugin and validates both mysql_native_password and caching_sha2_password master scrambles; a fixed SHA-2 protocol vector covers the new path. The focused 289-test RDS and protocol run passes.

**Review by pgermosen — APPROVED** [review]
Approving. Verified the crypto rather than trusting it - scrambleCachingSha2Password is XOR(SHA256(password), SHA256(SHA256(SHA256(password)), nonce)) exactly as MySQL's own docs specify, and extractMysqlAuthPlugin's byte offsets track the Handshake V10 packet layout field by field correctly. The image-tag parser handles every edge case I could think of, including the registry-port+patch+suffix+digest combination, which is already pinned in the tests. Solid, well-tested fix.
